lib/ansible/utils/hashing.py

```diff
@@ -31,7 +33,7 @@
 from ansible.module_utils.common.text.converters import to_bytes
 
 
-def secure_hash_s(data, hash_func=sha1):
+def secure_hash_s(data, hash_func=sha256):
```

Comment:
This is a breaking change for collections. Instead of changing the defaults for these functions, we should add new functions with the more secure default.

The old functions should then be deprecated -- or possibly even go through a deferred deprecation cycle (where we first mark them with `# deprecated` comments and then later deprecate them). The deferred deprecation cycle will be slower, but easier for collection maintainers as they should be able to avoid using fallback code to support older versions of ansible-core.